src/storage/src/source/postgres.rs

```diff
@@ -193,35 +194,30 @@ impl SourceRender for PostgresSourceConnection {
         // restart we are going to set the status to an ssh error correctly, so we don't do this
         // extra work.
 
-        let health = snapshot_err.concat(&repl_err).flat_map(move |err| {
+        let health = snapshot_err.concat(&repl_err).map(move |err| {
             // This update will cause the dataflow to restart
             let err_string = err.display_with_causes().to_string();
             let update = HealthStatusUpdate::halting(err_string.clone(), None);
-            let namespace = if matches!(
-                &*err,
-                TransientError::PostgresError(PostgresError::Ssh(_))
-                    | TransientError::PostgresError(PostgresError::SshIo(_))
-            ) {
-                StatusNamespace::Ssh
-            } else {
-                Self::STATUS_NAMESPACE.clone()
+
+            let namespace = match err {
+                // Is there a better way to reach into an Rc?
+                ReplicationError::Transient(err)
+                    if matches!(
+                        &*err,
+                        TransientError::PostgresError(PostgresError::Ssh(_))
+                            | TransientError::PostgresError(PostgresError::SshIo(_))
+                    ) =>
+                {
+                    StatusNamespace::Ssh
+                }
+                _ => Self::STATUS_NAMESPACE,
             };
-            let mut statuses = vec![HealthStatusMessage {
+
+            HealthStatusMessage {
                 index: 0,
                 namespace: namespace.clone(),
                 update,
-            }];
-
-            // But we still want to report the transient error for all subsources
-            statuses.extend(subsource_outputs.iter().map(|index| {
```

Comment:
You can remove the creation of `subsource_outputs` above now, since this was the only use-case for it

Reply:
Great catch, ty.